coalib/coala.py

```diff
@@ -71,10 +71,11 @@ def main():
     except BaseException as exception:  # pylint: disable=broad-except
         return get_exitcode(exception, log_printer)
 
-    if args.non_interactive:
+    if args.non_interactive and args.format:
+        return mode_format()
+    elif args.non_interactive:
         return mode_non_interactive(console_printer, args)
-
-    if args.format:
+    elif args.format:
         return mode_format()
```

Comment:
You would get exactly the same result by just switching the original check order to:

```python
if args.format:
    return mode_format()

if args.non_interactive:
    return mode_non_interactive(console_printer, args)
```